src/Query.php

```diff
@@ -454,7 +454,18 @@ public function assembleSelect()
         foreach ($resolved as $target) {
             $targetColumns = $resolved[$target]->getArrayCopy();
             if (isset($omitted[$target])) {
-                $targetColumns = array_diff($targetColumns, $omitted[$target]->getArrayCopy());
+                $toExclude = $omitted[$target]->getArrayCopy();
+                $targetColumns = array_filter($targetColumns, function ($value, $column) use ($toExclude) {
+                    if (is_string($column) && isset($toExclude[$column])) {
```

Comment:
Since when are the keys (integers) of ignored columns relevant?

Reply:
They are **not**. I'm not sure if I get you right, but your question and the diff context have actually the opposite effect.

Reply:
`$toExclude[$column]` can only be true if `$column` is a key in `$toExclude`. But `$toExclude` is a sequenced array and `$column` a string. This condition is always false.

Reply:
> But `$toExclude` is a sequenced array and `$column` a string.

Not if the column you want to exclude is an instance of `Expression`. You can remove the that if block and run the `testWithoutColumnsExcludesAliasedExpressions` unittest and see what happens.

Reply:
Oooh, because it's resolved already :man_facepalming: